tests/optim/test_oss.py

```diff
@@ -36,31 +35,16 @@
 DEVICE = "cuda" if torch.cuda.is_available() else torch.device("cpu")
 RECIPIENT_RANK = 1
 
-try:
-    from torch.distributed import broadcast_object_list  # noqa
-
-    _torch_broadcast_object = True
-except ImportError:
-    from fairscale.utils.params import broadcast_object  # noqa
-
-    _torch_broadcast_object = False
-
 
 def dist_init(rank, world_size, tempfile_name, backend=BACKEND):
     url = "file://" + tempfile_name
     dist.init_process_group(init_method=url, backend=backend, rank=rank, world_size=world_size)
 
 
 def sync_object_ranks(something_to_sync: Any, reference_rank: int, device: torch.device) -> Any:
-    if _torch_broadcast_object:
-        package = [something_to_sync]
-        dist.broadcast_object_list(package, src=reference_rank, group=dist.group.WORLD)
-        package_sync = package[0]
-    else:
-        package_sync = utils.params.broadcast_object(
-            something_to_sync, src_rank=reference_rank, group=dist.group.WORLD, dist_device=device
-        )
-
+    package = [something_to_sync]
```

Comment:
on the OSS side, I think that the best now would be to use the [distributed key value store](https://pytorch.org/docs/stable/distributed.html?highlight=key%20value%20store#distributed-key-value-store), in a non-blocking fashion. This could simplify the calls to .state_dict(), to my understanding, since all ranks could pull the state from the store and we could thus skip the .consolidate_state_dict(). I'm not completely sure that it's worth the work at this point in time, but just a heads up in case that's useful with FSDP

Reply:
Thanks. Sounds interesting. CC @zhaojuanmao  for her info as well.